enumfields/fields.py

```diff
@@ -32,7 +34,10 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else value.value
+        return None if value is None else self.enum(value).value
```

Comment:
Maybe we could do a type check here to avoid re-casting an already valid value?

```python
if value is None:
    return None
if not isinstance(value, self.enum):
    return self.enum(value).value
return value.value
```

or something? I benchmarked things a little and it looks like this saves quite some time in the common case (i.e. when an enum of the correct type is passed in).